e2e/tests/dump-restore-test.rs

```diff
@@ -181,61 +279,116 @@ fn dump_db(container: &Container<Cli, GenericImage>, db: &str, username: &str) -
         username,
         exit
     );
-
-    let src = PathBuf::from("/tmp/dump.sql");
-    let dest = env::temp_dir().join(format!("dump-{}.sql", db));
-    copy_out(container, &src, &dest);
-    dest
+    copy_out(container, dump, path);
 }
 
-fn restore_db(container: &Container<Cli, GenericImage>, db: &str, username: &str, path: PathBuf) {
-    copy_in(container, &path, &PathBuf::from("/tmp/dump.sql"));
-    psql_script(container, db, username, "/tmp/dump.sql");
+/// Restores a database from a logical backup generated by `pg_dump -F p`
+///
+/// # Arguments
+///
+/// * `container` - the docker container running postgres
+/// * `db` - the database to connect psql to with the `-d` flag to run the restore against
+/// * `username` - the user to run the restore script
+/// * `path` - the path on the host to the dump file
+///
+fn restore_db(container: &PostgresContainer, db: &str, username: &str, path: &Path) {
+    let dump = Path::new("/tmp/dump.sql");
+    copy_in(container, path, dump);
+    psql_file(container, db, username, dump);
 }
 
+/// Tests the process of dumping and restoring a database using pg_dump
+///
+/// 1. create a postgres docker container
+/// 2. create metrics, series, samples, exemplars, traces in the database
+/// 3. snapshot the database
+/// 4. use pg_dump to create a logical sql dump
+/// 5. destroy the docker container and create a new one
+/// 6. restore into the new database
+/// 7. snapshot the database
+/// 8. compare the two snapshots. they should be equal
+/// 9. compare one of the snapshots to a known good snapshot. they should be equal
 #[test]
 fn dump_restore_test() {
```

Comment:
This function is not very readable. Can be improved later.

Reply:
agreed. I have iterated on it a bit, but I'm not sure how much better it is

Reply:
resolved in [9263761](https://github.com/timescale/promscale_extension/pull/185/commits/92637615120da4f7db03e4c6c6adb2e08e5f66ca)